src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryLayouts.java

```diff
@@ -144,139 +144,6 @@ private MemoryLayouts() {
      */
     public static final ValueLayout JAVA_DOUBLE = MemoryLayout.ofValueBits(64, ByteOrder.nativeOrder());
 
-    /**
-     * The {@code _Bool} native type.
-     */
-    public static final ValueLayout C_BOOL;
-
-    /**
-     * The {@code unsigned char} native type.
-     */
-    public static final ValueLayout C_UCHAR;
-
-    /**
-     * The {@code signed char} native type.
-     */
-    public static final ValueLayout C_SCHAR ;
-
-    /**
-     * The {@code char} native type.
-     */
-    public static final ValueLayout C_CHAR;
-
-    /**
-     * The {@code short} native type.
-     */
-    public static final ValueLayout C_SHORT;
-
-    /**
-     * The {@code unsigned short} native type.
-     */
-    public static final ValueLayout C_USHORT;
-
-    /**
-     * The {@code int} native type.
-     */
-    public static final ValueLayout C_INT;
-
-    /**
-     * The {@code unsigned int} native type.
-     */
-    public static final ValueLayout C_UINT;
-
-    /**
-     * The {@code long} native type.
-     */
-    public static final ValueLayout C_LONG;
-
-    /**
-     * The {@code unsigned long} native type.
-     */
-    public static final ValueLayout C_ULONG;
-
-    /**
-     * The {@code long long} native type.
-     */
-    public static final ValueLayout C_LONGLONG;
-
-    /**
-     * The {@code unsigned long long} native type.
-     */
-    public static final ValueLayout C_ULONGLONG;
-
-    /**
-     * The {@code float} native type.
-     */
-    public static final ValueLayout C_FLOAT;
-
-    /**
-     * The {@code double} native type.
-     */
-    public static final ValueLayout C_DOUBLE;
-
-    /**
-     * The {@code T*} native type.
-     */
-    public static final ValueLayout C_POINTER;
-
-    static {
-        SystemABI abi = SystemABI.getInstance();
-        switch (abi.name()) {
```

Comment:
We could use `SystemABI::layoutFor` here to do the initialization instead though.

Reply:
Please see above. Per discussion, the idea is to keep C-independent layout constants in MemoryLayouts.java.